src/dotnet/ToolPackage/ToolPackageFactory.cs

```diff
@@ -21,13 +21,40 @@ internal static class ToolPackageFactory
             return (toolPackageStore, toolPackageInstaller);
         }
 
+        public static (IToolPackageStore, IToolPackageUninstaller) CreateToolPackageStoreAndUninstaller(
+            DirectoryPath? nonGlobalLocation = null)
+        {
+            IToolPackageStore toolPackageStore = CreateToolPackageStore(nonGlobalLocation);
+            var toolPackageUninstaller = new ToolPackageUninstaller(
+                toolPackageStore);
+
+            return (toolPackageStore, toolPackageUninstaller);
+        }
+
+        public static (IToolPackageStore,
+            IToolPackageInstaller,
+            IToolPackageUninstaller)
+            CreateToolPackageStoreInstallerUninstaller(
```

Comment:
Not a huge fan of this method name, but I can't suggest a better one.  Perhaps add an `And` for consistency with the one above?